### ISSUE TITLE
The propagated resources are not deleted when propagation policy is deleted

### ISSUE THREAD

**Issue by stingshen** [karmada-io/karmada#2380]
**What happened**:

When I delete a propagation policy, the applied resources are not deleted on the target clusters.

**What you expected to happen**:

The applied resources are deleted after I delete the propagation policy.

**How to reproduce it (as minimally and precisely as possible)**:

1. Create a deploy and a propagation policy that propagates the deploy to a cluster.
2. Wait for the deploy to be applied on the cluster.
3. Delete the propagation policy. And the deploy applied on step 2 is not deleted on the cluster.

**Anything else we need to know?**:

The resoure binding is not deleted either. Maybe we should set propagation policy to the ownerReferences of resource binding?

**Environment**: 
- Karmada version: v1.2.0
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`): 
```  
kubectl karmada version: version.Info{GitVersion:"v0.0.0-master", GitCommit:"unknown", GitTreeState:"unknown", BuildDate:"unknown", GoVersion:"go1.17.9", Compiler:"gc", Platform:"linux/amd64"}
```
- Others:

**Comment by stingshen** [comment]
The resoure binding is not  deleted either. Maybe we should set propagation policy to the ownerReferences of resource binding?

**Comment by RainbowMango** [comment]
> The applied resources are deleted after I delete the propagation policy.

As far as I know, this is the expected behavior when deleting a PropagationPolicy.

See here https://github.com/karmada-io/karmada/issues/403#issuecomment-852693968, you can remove the `deploy` from karmada control plane.

**Comment by stingshen** [comment]
> > The applied resources are deleted after I delete the propagation policy.
> 
> As far as I know, this is the expected behavior when deleting a PropagationPolicy.
> 
> See here [#403 (comment)](https://github.com/karmada-io/karmada/issues/403#issuecomment-852693968), you can remove the `deploy` from karmada control plane.

I agree, it is the expected behavior that the applied resources (e.g. deploy) are deleted from sub-clusters after the propagation policy is deleted.

But, the actual behavior now is the applied resources still exist on the sub-cluster after the propagation policy is deleted.

**Comment by RainbowMango** [comment]
Sorry, I didn't make myself clear. 
I mean this is the expected behavior, which means after the `PropagationPolicy` has been removed, the `ResourceBinding` and `Work` would keep existing, actually only the `PropagationPolicy` has been removed.

We have talked about this at #505 and implemented it by #601.

**Comment by stingshen** [comment]
Thank you for the explanation.
